### ISSUE TITLE
Preview images pulled from other sources are missing alt text even though the source image has alt text

### ISSUE THREAD

**Issue by ChasBelov** [mastodon/mastodon#37071]
### Steps to reproduce the problem

1. View [test post demonstrating the issue](https://ohai.social/@ChasMusic/115640794961394521)
2. Note the rendered preview image (as of November 30, 2025)
3. Follow the link to Wikipedia [Curb cut effect](https://en.wikipedia.org/wiki/Curb_cut_effect)
4. Right-click the corresponding image that matches the preview from step 2
5. Choose Inspect
6. Note the alt text for the image ("Sidewalk granite curb cut for wheelchair ramp" as of November 30, 2025)
7. Return to [test post demonstrating the issue](https://ohai.social/@ChasMusic/115640794961394521)
8. Right-click the preview image
9. Choose Inspect
...

### Expected behaviour

Preview image on Mastodon has same alt text as the Wikipedia source image

### Actual behaviour

Preview image on Mastodon has blank alt text

### Detailed description

When linking to another website, Mastodon pulls a sample image from the target page. However, it does not also pull the alt text from that image, an accessibility violation. I expect it to pull the alt text as well.

### Mastodon instance

ohai.social

### Mastodon version

v4.6.0-nightly.2025-11-25

### Browser name and version

Firefox 145.0.2

### Operating system

macOS 26.1

### Technical details

Code for image on Wikipedia site as of November 30, 2025:
```
<img alt="Sidewalk granite curb cut for wheelchair ramp" 
src="//upload.wikimedia.org/wikipedia/commons/thumb/b/bf/Curb_cut_for_wheelchair_ramp_%28DSC_3616%29.jpg/250px-Curb_cut_for_wheelchair_ramp_%28DSC_3616%29.jpg" 
decoding="async" width="250" height="167" class="mw-file-element" 
rcset="//upload.wikimedia.org/wikipedia/commons/thumb/b/bf/Curb_cut_for_wheelchair_ramp_%28DSC_3616%29.jpg/500px-Curb_cut_for_wheelchair_ramp_%28DSC_3616%29.jpg 1.5x" 
data-file-width="3000" data-file-height="2000">
```

Code for same image on Mastodon in my test post:
```
<img src="https://files.ohai.social/cache/preview_cards/images/000/088/186/original/856a9f79e65fc783.jpg" 
alt="" title="" lang="en" 
style="aspect-ratio: 1.91 / 1;" class="status-card__image-image">
```

**Comment by ClearlyClaire** [comment]
Hi! Thank you for your report!

The image is not pulled from the `img` tag, but from the following OpenGraph tag:
```html
<meta property="og:image" content="https://upload.wikimedia.org/wikipedia/commons/thumb/b/bf/Curb_cut_for_wheelchair_ramp_%28DSC_3616%29.jpg/1200px-Curb_cut_for_wheelchair_ramp_%28DSC_3616%29.jpg">
```

I'm not sure OpenGraph has any facility for providing an alt-text to such images. If there are, Mastodon does not support them, and Wikipedia does not make use of them either.

**Comment by ChasBelov** [comment]
According to the [Open Graph documentation for optional metadata](https://ogp.me/#optional):

> og:image:alt - A description of what is in the image (not a caption). If the page specifies an og:image it should specify og:image:alt.

So, if a webpage supplies og:image:alt, then Mastodon would properly render that as the alt text for the corresponding og:image source. I acknowledge that won't help with webpages that don't supply og:image:alt.

**Comment by ChasBelov** [comment]
Closing. I retested with a webpage that does supply og:image:alt and Mastodon renders the alt text.

**Comment by ClearlyClaire** [comment]
> Closing. I retested with a webpage that does supply og:image:alt and Mastodon renders the alt text.

Oh sorry, we do handle that, and I completely forgot about that.
